stko/optimizers/collapser.py

```diff
@@ -228,60 +231,571 @@ def optimize(self, mol):
+
+    def _get_bb_centroids(self, mol, bb_atom_ids):
+        """
+        Returns dict of building block centroids.
+
+        """
+
+        bb_centroids = {
+            i: mol.get_centroid(atom_ids=bb_atom_ids[i])
+            for i in bb_atom_ids
+        }
+
+        return bb_centroids
+
+    def _get_cent_to_lb_vector(
+        self,
+        mol,
+        bb_centroids,
+        long_bond_infos
+    ):
+        """
+        Returns dict of long bond atom to bb centroid vectors.
+
+        """
+
+        position_matrix = mol.get_position_matrix()
+        centroid_to_lb_vectors = {}
+        for bb in bb_centroids:
+            cent = bb_centroids[bb]
+            for b_atom_ids, bond_info in long_bond_infos.items():
+                for atom_id in b_atom_ids:
+                    atom_info, = mol.get_atom_infos(atom_ids=atom_id)
+                    atom_pos = position_matrix[atom_id]
+                    if atom_info.get_building_block_id() == bb:
+                        centroid_to_lb_vectors[(bb, atom_id)] = (
+                            atom_pos - cent,
+                        )
+                        break
+
+        return centroid_to_lb_vectors
+
+    def _bond_potential(self, distance):
+        """
+        Define an arbitrary parabolic bond potential.
+
+        This potential has no relation to an empircal forcefield.
+
+        """
+
+        potential = (distance - self._target_bond_length) ** 2
+        potential = self._bond_epsilon * potential
+        # force = -2 * (distance - self._target_bond_length)
+        # force = self._bond_epsilon * force
+
+        return potential
+
+    def _non_bond_potential(self, distance):
+        """
+        Define an arbitrary repulsive nonbonded potential.
+
+        This potential has no relation to an empircal forcefield.
+
+        """
+
+        return (
+            self._nonbond_epsilon * (
+                (self._nonbond_sigma/distance) ** self._nonbond_mu
+            )
+        )
+
+    def _compute_non_bonded_potential(self, mol):
+
+        # Get all pairwise distances.
+        pair_dists = pdist(mol.get_position_matrix())
+        non_bonded_potential = self._non_bond_potential(pair_dists)
+
+        return sum(non_bonded_potential)
+
+    def _compute_potential(self, mol, long_bond_infos):
+
+        system_potential = self._compute_non_bonded_potential(mol)
+        for long_bond_ids in long_bond_infos:
+            long_bond = long_bond_infos[long_bond_ids]
+
+            system_potential += self._bond_potential(
+                distance=self._get_bond_length(
+                    mol=mol,
+                    bond=long_bond.get_bond()
+                )
+            )
+
+        return system_potential
+
+    def _translate_atoms_along_vector(self, mol, atom_ids, vector):
+
+        new_position_matrix = mol.get_position_matrix()
+        for atom in mol.get_atom_infos(atom_ids=atom_ids):
+            id = atom.get_atom().get_id()
```

Comment:
`id` is an inbuilt Python function. Use `_id` as variable name instead as renaming this can cause  problems

Reply:
Fixed.